### PR TITLE
doc: bsim: Move contents into .rst

### PR DESCRIPTION
**Pull request by jori-nordic**
This makes the information more discoverable and linkable in the docs.

---
(Edit by @aescolar) Doc preview (when ready): https://builds.zephyrproject.io/zephyr/pr/70596/docs/develop/test/bsim.html

### CODE REVIEW

**Comment by jori-nordic** [comment]
I'm planning a series of doc improvements for bluetooth testing. This is the start of it.

File: doc/develop/test/bsim.rst [hunk 1]
```diff
@@ -76,3 +76,103 @@ Check :ref:`the page on coverage generation <coverage_posix>` for more info.
...
+
+   # Build all the tests
+   WORK_DIR=${ZEPHYR_BASE}/bsim_out ${ZEPHYR_BASE}/tests/bsim/compile.sh
+
+   # Run them (in parallel)
+   RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
+      SEARCH_PATH=${ZEPHYR_BASE}/tests/bsim \
+         ${ZEPHYR_BASE}/tests/bsim/run_parallel.sh
+
+Or to run only a specific subset, e.g. host advertising tests:
```
Comment (aescolar) [inline doc/develop/test/bsim.rst:107]:
```suggestion
Or to build and run only a specific subset, e.g. host advertising tests:
```

File: doc/develop/test/bsim.rst [hunk 2]
```diff
@@ -76,3 +76,103 @@ Check :ref:`the page on coverage generation <coverage_posix>` for more info.
...
+
+See the :ref:`nrf52_bsim` page for setting up the simulator.
+
+The scripts also expect a few environment variables to be set.
+For example, from Zephyr's root folder, you can run:
+
+.. code-block:: bash
+
+   # Build all the tests
+   WORK_DIR=${ZEPHYR_BASE}/bsim_out ${ZEPHYR_BASE}/tests/bsim/compile.sh
```
Comment (aescolar) [inline doc/develop/test/bsim.rst:100]: Sidenote: Setting `WORK_DIR` is optional nowadays (it defaults to `${ZEPHYR_BASE}/bsim_out`)

File: doc/develop/test/bsim.rst [hunk 3]
```diff
@@ -76,3 +76,103 @@ Check :ref:`the page on coverage generation <coverage_posix>` for more info.
...
+.. code-block:: bash
+
+   WORK_DIR=${ZEPHYR_BASE}/bsim_out ${ZEPHYR_BASE}/tests/bsim/net/compile.sh
+
+and then directly run one of the tests:
+
+.. code-block:: bash
+
+   ${ZEPHYR_BASE}/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh
+
```
Comment (aescolar) [inline doc/develop/test/bsim.rst:137]: On the reorder you seem to have missed the `Generating coverage reports` section?

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks @jori-nordic

File: doc/develop/test/bsim.rst (hunk 3 above)
Comment (jori-nordic) [reply doc/develop/test/bsim.rst:137]: https://github.com/jori-nordic/zephyr/blob/bsim-testing-rst/doc/develop/test/bsim.rst#L63
It's already here

File: doc/develop/test/bsim.rst [hunk 4]
```diff
@@ -76,3 +76,103 @@ Check :ref:`the page on coverage generation <coverage_posix>` for more info.
 
 .. _EDTT:
    https://github.com/EDTTool/EDTT
+
+Tests with radio activity
```
Comment (aescolar) [inline doc/develop/test/bsim.rst:80]: This header is now kinda repeating the one in L39

File: doc/develop/test/bsim.rst (hunk 2 above)
Comment (jori-nordic) [reply doc/develop/test/bsim.rst:100]: I removed it, its less verbose.

**Review by aescolar — DISMISSED** [review]

File: doc/develop/test/bsim.rst [hunk 5]
```diff
@@ -29,24 +29,28 @@ Types of tests
...
 
 When there is radio activity, BabbleSim tests require at the very least a physical layer simulation
 running, and most, more than 1 simulated device. Due to this, these tests are not build and run
 with twister, but with a dedicated set of tests scripts.
 
-These tests are kept in the :code:`tests/bsim/` folder. There you can find a README with more
-information about how to build and run them, as well as the convention they follow.
+These tests are kept in the :code:`tests/bsim/` folder. The ``compile.sh`` and
+``run_parallel.sh`` scripts contained in that folder are used by the CI system
+to build the needed images and execute these tests in batch.
```
Comment (alwa-nordic) [inline doc/develop/test/bsim.rst:48]: Use `tw` 100 to hard wrap text for consistency with the rest of the file.

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Approving, but on the condition that someone volunteers for joining me for a Zephyr Tech Talk about BabbleSim in the near future! :)
BabbleSim is definitely something I want to learn more about, and I am sure many others are in the same situation.

**Pull request merged**